### PR TITLE
Split Code into 2 repos, `climakitae` and `climakitaegui`.

### PR DESCRIPTION
**Pull request by elehmer**
# Description of PR

This PR is a major milestone in the `climakitae` development. It splits the code base into a base software package that will remain `climakitae` and a GUI package that will be in another repo `climakitaegui`. The base package will have a smaller requirements footprint so that more users can use the core functions of the software. The GUI package will contain anything related to `panel`, or the `holoviz` software universe that requires a greater amount of requirements.

See: https://github.com/cal-adapt/climakitaegui

At the moment I am unable to build a testing environment with just the base requirements. For the time being I am using the existing conda environment (conda-lock) until this can be sorted.

Documentation is in need of updating to reflect the changes to how the code is referenced (2 repos). API documentation needs to be vetted for which need to go into new gui repo.

**Summary of changes and related issue**

Many changes to import references occur because of code in 2 repos. Many classes have base class implementation and then are extended in the `climakitaegui` package.

**Relevant motivation and context**

Split of code base desired so that a minimal package with low requirements threshold can be distributed. This will allow users to use base functionality, such as data retrieval without the overhead of GUI code needed for the HUB.

**Dependencies required for this change?**

None, but `climakitaegui` will be needed on top of this package to run GUIs in HUB.

**Fixes # (issue), delete if not necessary**

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ x ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ x ] This change requires a documentation update

# How Has This Been Tested?

**Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.**

- [ x ] All notebooks have been tested on new codebase several times. There is a branch in cae-notebooks to support the needed changes to the notebooks: https://github.com/cal-adapt/cae-notebooks/tree/cksplit2a
- [ x ] Base package (this) has been tested at terminal python and able to run retrieval and export.

# Checklist:

- [ x ] My code follows the style guidelines of this project
- [ x ] I have performed a self-review of my code
- [ x ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ x ] New and existing unit tests pass locally with my changes
- [ x ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: climakitae/explore/warming.py [hunk 1]
```diff
@@ -12,6 +12,12 @@
 from climakitae.core.paths import (
     gwl_1981_2010_file,
     gwl_1850_1900_file,
+    ssp119_file,
```
Comment (elehmer) [inline climakitae/explore/warming.py:15]: @vicford These ssp files are only used in visualization and are imported in the warming code in `climakitaegui` so these are not needed here.

Comment (elehmer) [reply climakitae/explore/warming.py:15]: Ahh I see the confusion. I forgot to finish up with warming.py. Please review others for now and I will work on this.

Comment (vicford) [reply climakitae/explore/warming.py:15]: Sorry should have clarified @elehmer -- AMY breaks without them since it touches `warming.py`

Comment (elehmer) [reply climakitae/explore/warming.py:15]: Ahh, right. Hopefully will work when I rework warming

File: climakitae/core/data_export.py [hunk 2]
```diff
@@ -1144,15 +1144,9 @@ def write_tmy_file(
             Station longitude
         stn_state: str
             State of station location
-    <<<<<<< HEAD
```
Comment (elehmer) [inline climakitae/core/data_export.py]: @Tianchi-Liu  Thanks for catching that! Slipped through somehow

**Pull request merged**